p2p/protocol/holepunch/svc.go

```diff
@@ -172,6 +178,13 @@ func (s *Service) incomingHolePunch(str network.Stream) (rtt time.Duration, remo
 		return 0, nil, nil, fmt.Errorf("received hole punch stream: %s", str.Conn().RemoteMultiaddr())
 	}
 	ownAddrs = removeRelayAddrs(s.ids.OwnObservedAddrs())
+	interfaceListenAddrs, err := s.host.Network().InterfaceListenAddresses()
+
+	if err != nil {
+		log.Debugf("failed to get to get InterfaceListenAddresses: %s", err)
+	}
+	ownAddrs = ma.Unique(append(ownAddrs, interfaceListenAddrs...))
```

Comment:
We need to filter for public addresses in the list. 

I suggest making a method
```
 func (s *Service) getPublicAddrs() []ma.Multiaddr 
``` 

and using it in both the places.

Reply:
Addressed